src/common/snippets/src/lowered/pass/iter_handler.cpp

```diff
@@ -0,0 +1,108 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/lowered/pass/iter_handler.hpp"
+
+#include "snippets/itt.hpp"
+#include "snippets/lowered/linear_ir.hpp"
+#include "snippets/lowered/loop_manager.hpp"
+#include "snippets/lowered/pass/propagate_subtensors.hpp"
+#include "snippets/snippets_isa.hpp"
+#include "snippets/utils.hpp"
+
+namespace ov {
+namespace snippets {
+namespace lowered {
+namespace pass {
+UpdateMemoryAccessOps::UpdateMemoryAccessOps(size_t count) : RangedPass(), m_count(count) {}
+
+bool UpdateMemoryAccessOps::run(LinearIR& linear_ir, LinearIR::constExprIt begin, LinearIR::constExprIt end) {
+    for (auto expr_it = std::next(begin); expr_it != end; expr_it++) {
```

Comment:
Thanks for the suggestion. Yes, such behavior with first expression skipping must be avoided in passes. I managed to do that: now `InsertSpecificIterations` pass calls handlers on the range starting with the first expression in loop body (to avoid LoopBegin handling inside each pass)